apps/api/tests/test_models.py

```diff
@@ -17,6 +19,35 @@ def test_new_jwt_credentials(self):
         assert credentials.type == SYMMETRIC_JWT_TYPE
         assert credentials.key
         assert credentials.secret
+        assert credentials.is_active
+
+    def test_string_representation(self):
+        credentials = APIKey.new_jwt_credentials(self.user)
+        str_creds = str(credentials)
+        assert credentials.key in str_creds
+        assert credentials.secret not in str_creds
+        assert str(credentials.user) in str_creds
+
+    def test_generate_new_unique_keys(self):
+        last_key = None
+        for counter in range(3):
+            credentials = APIKey.new_jwt_credentials(self.user)
+            assert credentials.key != last_key
+            last_key = credentials.key
+
+    def test_too_many_tries_at_finding_a_unique_key(self):
+        max = 3
+
+        # Make APIKey.objects.filter().exists() always return True.
+        patch = mock.patch('apps.api.models.APIKey.objects.filter')
+        mock_filter = patch.start()
+        self.addCleanup(patch.stop)
+        mock_filter.return_value = mock.Mock()
+        mock_filter.return_value.exists = mock.Mock(return_value=True)
```

Comment:
I don't think you need to explicitly call `mock.Mock()`. You should be able to do `mock_filter.return_value.exists.return_value = True`.